dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+# coding: utf-8
+"""Dragonfly Room Polygon."""
+from ._base import _Base
+from .properties import RoomPolygonProperties
+from .glazingparameter import GlazingParameter
+
+import honeybee.boundarycondition.boundary_conditions as bcs
+from honeybee.boundarycondition import _BoundaryCondition, Outdoors, Surface
+from honeybee.typing import tuple_with_length, float_positive
+from honeybee.room import Room
+from honeybee.aperture import Aperture
+
+from ladybug_geometry.geometry3d.face import Face3D
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+
+import math
+
+
+class RoomPolygon(_Base):
+    """A volume defined by an extruded floor plate, representing a single room or space.
+
+    Properties:
+        * name
+        * display_name
+        * geometry
+        * floor_to_ceiling_height
+        * boundary_conditions
+        * glazing_ratios
+        * glazing_parameters
+        * parent
+        * has_parent
+        * floor_segments
+        * segment_count
+        * floor_height
+        * ceiling_height
+        * volume
+        * floor_area
+        * exterior_wall_area
+        * exterior_aperture_area
+    """
+    __slots__ = ('_geometry', '_segment_count', '_floor_to_ceiling_height',
+                 '_boundary_conditions', '_glazing_ratios', '_glazing_parameters',
+                 '_parent')
+
+    def __init__(self, name, geometry, floor_to_ceiling_height, boundary_conditions=None,
+                 glazing_ratios=None, glazing_parameters=None, tolerance=None):
+        """A volume defined by an extruded floor plate, representing a single room.
+
+        Args:
+            name: Room name. Must be < 100 characters.
+            geometry: A single Face3D object representing the floor plate of the Room.
```

Comment:
Ok. I can add `classmethods `for `from_vertices` and `from_polygon2d`. These cannot be the default `__init__` method, though, since you can't tell a hole from a boundary in a single list of vertices and I know we want to be able to support things like courtyards in Dragonfly. Also, [Polygons](https://en.wikipedia.org/wiki/Simple_polygon) by definition do not have holes in them and it is for this reason that the Polygon2D class in ladybug-geometry does not support holes (so the `from_polygon` method suffers from the same limitation as the `from_vertices` method).
Maybe down the line we will add a `PSLG` class (short for [Planar Straight-Line Graph](https://en.wikipedia.org/wiki/Planar_straight-line_graph)) to ladybug-geometry, which is the technical term for a 2D shape that can support holes. For now, though, it seems the simplest and most elegant solution in Dragonfly is to use horizontal `Face3D` objects to represent the room polygon.

Reply:
I added the classmethods.